### PR TITLE
Proto api

### PR DESCRIPTION
**Pull request by benja-wu**
* Complete first version's EaseMesh proto API define.
* This commit doesn't include `ServiceInstance` spec, coming at next PR.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module easemesh-api
```
Comment (zhao-kun) [inline go.mod]: Why not full qulified repository url?

File: v1alpha1/meshmodel.proto [hunk 2]
```diff
@@ -229,11 +260,225 @@ message Resilience {
...
+    ObservabilityTracings  tracings = 2;
+
+    // Metrics configures the metrics JavaAgent should collect. 
+    ObservabilityMetrics    metrics = 3;
+}
+
+// ObservabilityOutputServer configures how to report observability data to Kafka.
+message ObservabilityOutputServer {
+    // Enables configures to report observability data to Kafka or not.
+    bool Enabled  = 1;
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: Why word `enabled` first letter is capitalize ?

Comment (benja-wu) [reply v1alpha1/meshmodel.proto]: It's a typo, Let me fix it....

File: go.mod (hunk 1 above)
Comment (benja-wu) [reply go.mod]: According to Istio's `module istio.io/api`, is that `module megaease/easemesh-api` qulified enough?

Comment (benja-wu) [reply go.mod]: I will update it to `module github.com/megaease/easemesh-api` according to our EG project's module.

Comment (zhao-kun) [reply go.mod]: following Golang's best practice, refer to our repos.

File: v1alpha1/meshmodel.proto [hunk 3]
```diff
@@ -229,11 +260,225 @@ message Resilience {
...
+    // CanaryRules is the mesh service's all rules for canary deployment. 
+    repeated CanaryRule canaryRules = 1;
+}
+
+// Sidecar configures the sidecar for every mesh service instances. It works inside the same
+// pod with the workload instance. 
+message Sidecar {
+    // DiscoveryType configures the type of service register/discovery type, its value are among 
+    // "eureka","consul", and "nacos".
+    string discoveryType = 1;
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: three spaces?

File: v1alpha1/meshmodel.proto [hunk 4]
```diff
@@ -229,11 +260,225 @@ message Resilience {
...
+    // IngressPort is the port for ingress traffic.
+	int32 ingressPort = 3;
+
+    // IngressProtocol is the protocol for ingress traffic. Its value is "http".
+	string ingressProtocol = 4;
+
+    // egressPort is the port for egress traffic.
+	int32 egressPort = 5;
+
+    // egressProtocol is the protocol for egress traffic. Its value is "http"
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: 1. Align indentation of comment to code
2. Comment first word should be same with code

File: v1alpha1/meshmodel.proto [hunk 5]
```diff
@@ -229,11 +260,225 @@ message Resilience {
...
+	string egressProtocol = 6;
+}
+
+// LoadBalance configures how to distribute the traffic inside this mesh.
+message LoadBalance {
+    // Policy including four kinds of load balancing scheme, including random, weightedRandom,ipHash,
+    // headerHash
+    string policy = 1;
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: // policy ....

File: v1alpha1/meshmodel.proto [hunk 6]
```diff
@@ -229,11 +260,225 @@ message Resilience {
...
+    // Tracings configures whether JavaAgent should deal with tracing HTTP header or not. 
+    ObservabilityTracings  tracings = 2;
+
+    // Metrics configures the metrics JavaAgent should collect. 
+    ObservabilityMetrics    metrics = 3;
+}
+
+// ObservabilityOutputServer configures how to report observability data to Kafka.
+message ObservabilityOutputServer {
+    // Enables configures to report observability data to Kafka or not.
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: // enabled

File: v1alpha1/meshmodel.proto [hunk 7]
```diff
@@ -229,11 +260,225 @@ message Resilience {
...
+    // BootstrapServer configures the Kafka bootstrap address.
+    string bootstrapServer = 2;
+
+    // Timeout configures the million second for Kafka's timeout. 
+    int32 timeout = 3; 
+}
+
+// ObservabilityTracingsDetail 
+message ObservabilityTracingsDetail {
+    // Enabled configures to enable this tracing component or not.
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: // enabled

**Review by zhao-kun — COMMENTED** [review]
There are many inconsistent comments, please carefully check and fixed it

File: v1alpha1/meshmodel.proto [hunk 8]
```diff
@@ -210,12 +226,27 @@ message Service {
 
     // registerTenant is the tenant's name this service belongs to.
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: Please ensure all first letter of first word in comment line is captilize!

File: v1alpha1/meshmodel.proto [hunk 9]
```diff
@@ -229,11 +260,225 @@ message Resilience {
...
+    // Address is the registry center address for workload to visit. 
+	string address = 2; 
+
+    // IngressPort is the port for ingress traffic.
+	int32 ingressPort = 3;
+
+    // IngressProtocol is the protocol for ingress traffic. Its value is "http".
+	string ingressProtocol = 4;
+
+    // egressPort is the port for egress traffic.
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: Please ensure all first letter of the first word in comment line is capitilize!

**Review by zhao-kun — APPROVED** [review]

**Pull request merged**